src/components/Facility/FacilityCover.tsx

```diff
@@ -0,0 +1,91 @@
+import { navigate } from "raviger";
+import { useEffect } from "react";
+
+import { Cancel, Submit } from "@/components/Common/ButtonV2";
+import Page from "@/components/Common/Page";
+import { CoverImage } from "@/components/Facility/CoverImage";
+
+import useAuthUser from "@/hooks/useAuthUser";
+
+import * as Notification from "@/Utils/Notifications";
+import routes from "@/Utils/request/api";
+import useTanStackQueryInstead from "@/Utils/request/useQuery";
+
+interface FacilityProps {
+  facilityId: string;
+}
+
+export const FacilityCover = (props: FacilityProps) => {
+  const { facilityId } = props;
+
+  const { data: facilityData } = useTanStackQueryInstead(
+    routes.getPermittedFacility,
+    {
+      pathParams: {
+        id: facilityId,
+      },
+      onResponse: ({ res }) => {
+        if (!res?.ok) {
+          navigate("/not-found");
+        }
+      },
+    },
+  );
```

Comment:
use useQuery instead

https://docs.ohc.network/docs/care/development/data-fetching-in-care/

Reply:
done!!!
@rithviknishad

Reply:
review the code yourself :) there are more usages of it

Reply:
@rithviknishad 
Yeah, I saw that but the code requires reFetch, is there a way to destructure refetch out of useQuery?

Reply:
you can destructure `refetch` :) from usequery.

```
const { refetch } = useQuery(...);
```

however, the cleaner way to trigger the refetch is by invalidating the query key instead of calling refetch.

Reply:
sorry, I couldn't quiet get you

Reply:
https://tanstack.com/query/latest/docs/reference/QueryClient#queryclientinvalidatequeries

Reply:
can you check now